File: arrays/tournamentWinner/tournament.py
#alternate solutions


def tournamentWinner(comp, results):

dic = {} #dictionary to store winners
max = 0 #variable to track min points of any team
current_winner = '#' #variable to track current winner

for i in range(len(comp)): #iterate through comp list

  match = comp[i]
  resultIndex = abs(results[i] - 1) #find index of winner from result list
  winner = comp[i][resultIndex]

  if winner in dic:
      pts = dic[winner]
      pts += 3
      if pts > max: #check to see if update pts is greater than current maximum
          max = pts
          current_winner = winner
      dic.update({winner : pts}) #update winner record

  else:
      pts = 3
      dic[winner] = 3
      if pts > max: #check to see if winner pts is greater than current maximum
          max = pts
          current_winner = winner

return current_winner

File: arrays/tournamentWinner/tournament.js
const home_team_win = 1;

function tournamentWinner (competitions, results) {
  let current_best_team = '';
  const scores = {[current_best_team]: 0};

  for (let i = 0; i < competitions.length; i++) {
    const result = results[i];
    const [homeTeam, awayTeam] = competitions[i];

    const winningTeam = result === home_team_win ? homeTeam : awayTeam;

    if (scores[winningTeam] > scores[current_best_team]) {
      current_best_team = winningTeam;
    }
  }
  return current_best_team;
}
